dulwich/tests/test_porcelain.py

```diff
@@ -1375,3 +1376,44 @@ def test_git_dir(self):
         self.assertEqual(
                 [(obj.id, 'invalid name .git')],
                 [(sha, str(e)) for (sha, e) in porcelain.fsck(self.repo)])
+
+
+class HelperTests(TestCase):
+
+    def test_path_to_tree_path_base(self):
+        self.assertEqual(b"bar/baz", porcelain.path_to_tree_path(
+            "/home/foo", "/home/foo/bar/baz"))
+        self.assertEqual(b"bar/baz", porcelain.path_to_tree_path(
+            ".", "./bar/baz"))
+        self.assertEqual(b"bar/baz", porcelain.path_to_tree_path(
+            ".", "bar/baz"))
+        cwd = os.getcwd()
+        path = os.path.join(cwd, "bar/baz")
+        self.assertEqual(b"bar/baz", porcelain.path_to_tree_path(
+            ".", path))
+        self.assertEqual(b"bar/baz", porcelain.path_to_tree_path(
+            cwd, "bar/baz"))
+        os.mkdir("foo")
```

Comment:
Please use TestCaseInTempDir if you're going to create files/directories, otherwise we're leaving these all over the source directory.

Reply:
My bad, Will change first thing tomoreiw